freeipa/src/test/java/com/sequenceiq/freeipa/flow/stack/upgrade/ccm/UpgradeCcmFlowIntegrationTest.java

```diff
@@ -214,16 +222,39 @@ private void verifyServiceCalls(int calledOnceCount) throws CloudbreakOrchestrat
         inOrder.verify(upgradeCcmService, times(expected[i++])).changeTunnel(STACK_ID);
         inOrder.verify(upgradeCcmService, times(expected[i++])).obtainAgentDataState(STACK_ID);
         inOrder.verify(upgradeCcmService, times(expected[i++])).obtainAgentData(STACK_ID);
+        if (calledOnceCount == CALLED_ONCE_TILL_OBTAIN_AGENT_DATA) {
```

Comment:
Can you please guide me through this?

Reply:
Yes, It is not the most elegant,but these are the revert cases for every cases. so the test will pass calledOnceCount meaning this is where exception will happen, and (after this all the call count will be 0) checking the calls for the revert branch.